cpp/src/ice2slice/Gen.cpp

```diff
@@ -12,26 +12,54 @@ using namespace IceInternal;
 
 namespace
 {
-    static string getCSharpNamespace(const ContainedPtr& cont, bool& hasCSharpNamespaceAttribute)
+    static void writeCsIdentifier(const ContainedPtr& cont, Output& out)
+    {
+        // Modules are not allowed to have a cs::identifier attribute in IceRPC Slice.
+        // they are handled separately in the getOutput function.
+        assert(!dynamic_pointer_cast<Module>(cont));
+
+        if (auto metadata = cont->getMetadataArgs("cs:identifier"))
+        {
+            out << nl << "[cs::identifier(\"" << *metadata << "\")]";
+        }
+    }
+
+    static string getCSharpMappedName(const ContainedPtr& cont)
+    {
+        if (auto metadata = cont->getMetadataArgs("cs:identifier"))
+        {
+            return *metadata;
+        }
+        else
+        {
+            return cont->name();
+        }
+    }
+
+    static string getCSharpMappedScoped(const ContainedPtr& cont, const string& separator);
```

Comment:
We cannot use mappedName, mappedScope, .. because they depend on the language-mapping set when you create the Unit.

Reply:
Would be nice to add this comment in the code.

Also, what about calling these function getCsXxx, etc. Having both cs and csharp looks odd.

Reply:
Or just call it `getMappedScoped` and have it take a language prefix parameter.
In the future, when we add Python support for icerpc, we'd want to re-use these for python metadata as well.

Reply:
We can rename them when we add support for new mappings, all this code is internal to ice2slice.